xdsl/dialects/experimental/air.py

```diff
@@ -0,0 +1,810 @@
+        region: Region | None,
+    ):
+        super().__init__(
+            attributes={"sym_name": sym_name},
+            operands=[async_dependencies, sizes, herd_operands],
+            result_types=[AsyncTokenAttr()],
+            regions=[region],
+        )
+
+    def print(self, printer: Printer):
+        printer.print(" tile")
+        printer.print("(")
+        if len(self.sizes) == 1:
+            printer.print("%tx")
+        if len(self.sizes) == 2:
+            printer.print("%tx, %ty")
+        if len(self.sizes) == 3:
+            printer.print("%tx, %ty, %tz")
+        printer.print(")")
+        printer.print(" in ")
+        printer.print("(")
+        if len(self.sizes) == 1:
+            printer.print("%\\size_x = ")
+        if len(self.sizes) == 2:
+            printer.print("%size_x = ")
+            printer.print(self.sizes[0])
+            printer.print(", ")
+            printer.print("%size_y = ")
+            printer.print(self.sizes[1])
+        if len(self.sizes) == 3:
+            printer.print("%\\size_x, %\\size_y, %\\size_z")
+        printer.print(")")
+
+        if self.herd_operands:
+            printer.print(" args")
+            printer.print("(")
+            if len(self.sizes) == 1:
+                printer.print("%\\ext0 = ")
+            if len(self.sizes) == 2:
+                printer.print("%ext0 = ")
+                printer.print(self.herd_operands[0])
+                printer.print(", ")
+                printer.print("ext1 = ")
+                printer.print(self.herd_operands[1])
+            if len(self.sizes) == 3:
+                printer.print("%ext0 = ")
+
+        printer.print(" : ")
+        for n_arg in range(len(self.herd_operands)):
+            printer.print(self.herd_operands[n_arg].type)
+            if n_arg < len(self.herd_operands) - 1:
+                printer.print(",")
+
+        printer.print(")")
+        if self.region:
+            printer.print_region(self.region)
+
+    @classmethod
+    def parse(cls, parser: Parser) -> HerdOp:
+        sym_name = parser.parse_optional_symbol_name()
+        async_dependencies: list[Operation | SSAValue] = []
+        if parser.parse_optional_keyword("async"):
+            parser.parse_optional_characters("[")
+            while not parser.parse_optional_characters("]"):
+                async_dependencies.append(parser.parse_operand())
+                parser.parse_optional_characters(",")
+        parser.parse_keyword("tile")
+        arg_list = parser.parse_op_args_list()
+        parser.parse_keyword("in")
+
+        parser.parse_characters("(")
+        tile_size_lst: list[Operation | SSAValue] = []
+        for n_arg in range(len(arg_list)):
+            parser.parse_optional_argument(False)
+            parser.parse_characters("=")
+            tile_size = parser.parse_operand()
+            tile_size_lst.append(tile_size)
+            if n_arg < len(arg_list) - 1:
+                parser.parse_characters(",")
```

Comment:
Could use `parser.parse_comma_separated_list` too, with a helper for this `op = op` thing.
Also `parser.parse_optional_argument(False)` is just `parser.parser_optional_operand`

Reply:
Is this helper something that might be interesting to add to the core of xDSL? I don't know how common this custom format is out there